### ISSUE TITLE
Bulk load Script mismatching with Tinkerpop 3.3.x

### ISSUE THREAD

**Issue by boliza** [JanusGraph/janusgraph#1102]
`parse` method  signature has change from `def parse(line, factory)` to `def parse(line)`

Script files:
1. [legacy-graphson-script-input.groovy](https://github.com/JanusGraph/janusgraph/blob/master/janusgraph-dist/src/assembly/static/data/legacy-graphson-script-input.groovy)
2. [script-input-grateful-dead.groovy](https://github.com/JanusGraph/janusgraph/blob/master/janusgraph-dist/src/assembly/static/data/script-input-grateful-dead.groovy)
3. [script-input-tinkerpop.groovy](https://github.com/JanusGraph/janusgraph/blob/master/janusgraph-dist/src/assembly/static/data/script-input-tinkerpop.groovy)

See [Script IO format](http://tinkerpop.apache.org/docs/3.3.2/reference/#script-io-format)

See [READ_CALL](https://github.com/apache/tinkerpop/blob/93a64349b2ca9fa9da0c7a0bd33d3b44a436ffbc/hadoop-gremlin/src/main/java/org/apache/tinkerpop/gremlin/hadoop/structure/io/script/ScriptRecordReader.java#L63)

**Comment by pluradj** [comment]
thanks for the report @boliza. will you be submitting a PR for this?

**Comment by boliza** [comment]
Sorry to reply later. I would try to fix this.

But the problem I described on [janusgraph-users](https://groups.google.com/forum/#!topic/janusgraph-users/RO8FyqxDiXM),seems a bug of Tinkerpop

Code in `BulkLoaderVertexProgram`

```java
final Map<Object, Object> idPairs = new HashMap<>();
final Iterator<Tuple> idi = messenger.receiveMessages();
while (idi.hasNext()) {
    final Tuple idPair = idi.next();
    idPairs.put(idPair.getValue(0), idPair.getValue(1));
}
```

In my case, `messenger` has no `receiveMessages()`

So I had fix like this:

```java
Object inVId = idPairs.get(edge.inVertex().id());
if (inVId == null) {
     inVId = bulkLoader.getByBulkLoaderVertexId(edge.inVertex().id(), g);
}
```

**Comment by boliza** [comment]
I test `script-input-grateful-dead.groovy` .  

```gremlin
gremlin> graph = GraphFactory.open('conf/hadoop-graph/hadoop-script.properties')
==>hadoopgraph[scriptinputformat->graphsonoutputformat]
gremlin> blvp = BulkLoaderVertexProgram.build().writeGraph('conf/janusgraph-cql-es.properties').create(graph)
==>BulkLoaderVertexProgram[bulkLoader=IncrementalBulkLoader, vertexIdProperty=bulkLoader.vertex.id, userSuppliedIds=false, keepOriginalIds=true, batchSize=0]
gremlin> graph.compute(SparkGraphComputer).program(blvp).submit().get()
12:32:35 WARN  org.apache.spark.util.Utils  - Your hostname, tsao resolves to a loopback address: 127.0.1.1; using 192.168.10.29 instead (on interface wlp3s0)
12:32:35 WARN  org.apache.spark.util.Utils  - Set SPARK_LOCAL_IP if you need to bind to another address
==>result[hadoopgraph[scriptinputformat->graphsonoutputformat],memory[size:0]]
gremlin> graph.close()
gremlin> graph = JanusGraphFactory.open('conf/janusgraph-cql-es.properties')
==>standardjanusgraph[cql:[127.0.0.1]]
gremlin> g = graph.traversal()
gremlin> g.V().count()
12:35:55 WARN  org.janusgraph.graphdb.transaction.StandardJanusGraphTx  - Query requires iterating over all vertices [()]. For better performance, use indexes
==>808
```

So seems if I do not custom `bulkLoader.vertex.id`, will be bulk load sucessful
